fastserde/avro-fastserde-tests111/build/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_11/Array_of_TestRecord_SpecificDeserializer_1088113243_1088113243.java

```diff
@@ -527,39 +528,47 @@ private void populate_TestRecord11(com.linkedin.avro.fastserde.generated.avro.Te
                 if (oldArray4 instanceof GenericArray) {
                     recordsArrayArrayElementReuseVar0 = ((GenericArray) oldArray4).peek();
                 }
-                recordsArray0 .add(deserializeSubRecord0(recordsArrayArrayElementReuseVar0, (decoder)));
+                recordsArray0 .add(deserializeSubRecord0(recordsArrayArrayElementReuseVar0, (decoder), (customization)));
             }
             chunkLen5 = (decoder.arrayNext());
         }
         TestRecord.put(24, recordsArray0);
     }
 
-    private void populate_TestRecord12(com.linkedin.avro.fastserde.generated.avro.TestRecord TestRecord, Decoder decoder)
+    private void populate_TestRecord12(com.linkedin.avro.fastserde.generated.avro.TestRecord TestRecord, DatumReaderCustomization customization, Decoder decoder)
         throws IOException
     {
         Map<Utf8, com.linkedin.avro.fastserde.generated.avro.SubRecord> recordsMap0 = null;
         long chunkLen6 = (decoder.readMapStart());
         if (chunkLen6 > 0) {
-            Map<Utf8, com.linkedin.avro.fastserde.generated.avro.SubRecord> recordsMapReuse0 = null;
-            Object oldMap0 = TestRecord.get(25);
-            if (oldMap0 instanceof Map) {
-                recordsMapReuse0 = ((Map) oldMap0);
-            }
-            if (recordsMapReuse0 != (null)) {
-                recordsMapReuse0 .clear();
-                recordsMap0 = recordsMapReuse0;
+            if (((customization)!= null)&&((customization).getNewMapOverrideFunc()!= null)) {
```

Comment:
Hmm... we know at code gen time whether these objects are null, right? It would be nice to eliminate this fork in the generated code, and generate only the path that will actually execute...

Or another way to do it might be that at code gen time, if these objects are null, then we populate them with a default impl which is essentially the same as the `else` branch below, then the generated code always calls the customization function.

LMK what you think...